packages/messaging/src/controllers/controller-interface.ts

```diff
@@ -113,26 +113,35 @@ export default class ControllerInterface {
   }
 
   /**
-   * This method deletes tokens that the token manager looks after and then
-   * unregisters the push subscription if it exists.
+   * This method deletes tokens that the token manager looks after,
+   * unsubscribes the token from FCM  and then unregisters the push
+   * subscription if it exists. It returns a promise that indicates
+   * whether or not the unsubscribe request was processed successfully.
    * @export
-   * @param {string} token
-   * @return {Promise<void>}
    */
-  deleteToken(token) {
-    return this.tokenDetailsModel_.deleteToken(token).then(() => {
-      return this.getSWRegistration_()
-        .then(registration => {
-          if (registration) {
-            return registration.pushManager.getSubscription();
-          }
-        })
-        .then(subscription => {
-          if (subscription) {
-            return subscription.unsubscribe();
-          }
-        });
-    });
+  deleteToken(token: string): Promise<Boolean> {
+    return this.tokenDetailsModel_
+      .deleteToken(token)
+      .then(details => {
+        return this.iidModel_.deleteToken(
```

Comment:
Will the application server key be needed to delete this as well, or is that more a piece of meta data that goes along with this info?

Reply:
nope, it shouldn't be needed - app pub key is treated the same way as the PushSubscription in this case.